test/integration/jobset_controller_test.go

```diff
@@ -71,18 +70,17 @@ var _ = ginkgo.Describe("JobSet controller", func() {
 	// jobSetUpdate contains the mutations to perform on the jobset and the
 	// checks to perform afterwards.
 	type jobSetUpdate struct {
-		jobUpdateFn             func(*batchv1.JobList)
-		checkJobSetState        func(*jobset.JobSet)
-		expectedJobSetCondition jobset.JobSetConditionType
+		updateFn            func(*jobset.JobSet)
+		updateShouldSucceed bool
```

Comment:
sorry, another nit: we can remove the update prefix for both parameters, the struct name is update already